### ISSUE TITLE
3.0.3 broke `async` handler

### ISSUE THREAD

**Issue by mattbooks** [getsentry/raven-ruby#1001]
Async handlers call send_event with a hash: 

https://github.com/getsentry/raven-ruby/blob/a0d09f6b9a8533cb749f4ec7a71bae25882db8a0/lib/raven/instance.rb#L123

https://github.com/getsentry/raven-ruby/blob/master/lib/raven/event.rb#L136

The refactor of Client now assumes an object:

https://github.com/getsentry/raven-ruby/blob/master/lib/raven/client.rb#L38

This breaks users of `async`, including the example in the readme:

```
config.async = lambda { |event|
  Thread.new { Raven.send_event(event) }
}
```

This was broken in the refactor change #995 

cc @st0012

**Comment by agrobbin** [comment]
We've run into this as well, after Dependabot bumped us from 3.0.2 to 3.0.3. All of our Sentry background jobs started raising exceptions, so we have temporarily reverted back to 3.0.2.

If it helps, the issue manifests as a couple of different exceptions:

```
NoMethodError: undefined method `event_id' for #<Hash:0x00007f06cc8ec1a8>
NoMethodError: undefined method `log_message' for #<Hash:0x00007f06ce260648>
```

**Comment by st0012** [comment]
thanks for reporting and sorry for all the hassles 😓  
I've added https://github.com/getsentry/raven-ruby/pull/1002 to revert the changes & increase test coverage.

**Comment by st0012** [comment]
The fix has went out in version 3.0.4

**Comment by agrobbin** [comment]
Thanks for the quick fix and release @st0012!
